### PR TITLE
Add support for Travis Pro

### PR DESCRIPTION
**Pull request by jacobwinch**
The Travis support in Prout currently only works with Travis CI. However, we want to use Prout to run post-deployment tests for a private project, which means that we need to add compatibility for the Travis Pro API.

There's some more information on the different API endpoints here:
https://docs.travis-ci.com/api

This PR allows Prout to call the correct API, based on the private/public status of the repo being scanned.

### CODE REVIEW

File: app/lib/travis/TravisApiClient.scala [hunk 1]
```diff
@@ -19,114 +18,147 @@ class TravisApiClient(githubToken: String) extends LazyLogging {
...
+
+    object AuthResponse {
+      implicit val readsAuthRequest = Json.reads[AuthResponse]
+    }
+
+    def auth(githubToken: String) = postJson[AuthRequest, AuthResponse]("/auth/github", AuthRequest(githubToken))
 
-    for {
+    def build(buildId: String) = for {
       authToken <- authTokenSupplier.get()
```
Comment (jacobwinch) [inline app/lib/travis/TravisApiClient.scala]: This request doesn't _require_ auth headers for the Travis CI API, but it does for the Travis Pro API. As we're using token refresh anyway, we just pass the auth header regardless of the environment.

File: conf/application.conf [hunk 2]
```diff
@@ -17,6 +17,11 @@ github {
   clientSecret=${?GITHUB_APP_CLIENT_SECRET}
 }
 
+travis.token {
+  opensource=${?PROUT_TRAVIS_TOKEN_OPENSOURCE}
+  commercial=${?PROUT_TRAVIS_TOKEN_COMMERCIAL}
```
Comment (rtyley) [inline conf/application.conf]: @jacobwinch we'll need to generate a Travis CI token for `travis-ci.com` and set it as a  `PROUT_TRAVIS_TOKEN_COMMERCIAL` environment variable for the Prout instance.

File: build.sbt [hunk 3]
```diff
@@ -31,7 +31,7 @@ libraryDependencies ++= Seq(
   "com.typesafe.scala-logging" %% "scala-logging" % "3.5.0",
   "com.github.scala-incubator.io" %% "scala-io-file" % "0.4.3-1",
   "org.webjars.bower" % "octicons" % "3.1.0",
-  "com.madgag" %% "play-git-hub" % "4.4",
+  "com.madgag" %% "play-git-hub" % "4.5",
```
Comment (rtyley) [inline build.sbt]: Adds support for reading *all* commit statuses on a branch - not just the latest commit status: https://github.com/rtyley/play-git-hub/commit/92fee70d

File: app/lib/travis/TravisApiClient.scala [hunk 4]
```diff
@@ -92,41 +104,27 @@ class TravisApiClient(githubToken: String) extends LazyLogging {
...
 
-    /**
-     * If no refresh is currently running, start a new one, return that future
-     * If a refresh is currently running, return the existing future
-     */
-    def refresh(): Future[T] = for {
-      refreshAlteringFuture <- agent.alter { currentRefresh => if (currentRefresh.isCompleted) generateToken else currentRefresh }
-      refreshFuture <- refreshAlteringFuture
-    } yield refreshFuture
+object TravisCIOffering {
```
Comment (rtyley) [inline app/lib/travis/TravisApiClient.scala]: The `OpenSource` & `Commercial` objects below are the _"stateless objects that just encapsulate the API url"_ that I was talking about in https://github.com/guardian/prout/pull/57#discussion_r156678804 - they don't have state, they just do the sole job of expressing the difference between `travis-ci.org` (free) & `travis-ci.com` (commercial) offerings.

**Pull request merged**